feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go

```diff
@@ -0,0 +1,736 @@
+func pushConfig(t *testing.T, dut *ondatra.DUTDevice, dutPort *ondatra.Port, d *oc.Root) {
+	t.Helper()
+
+	intfName := dutPort.Name()
+	i := d.GetOrCreateInterface(intfName)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(intfName).Config(), i)
+}
+
+// configureInterfaceDUT configures a single DUT layer 2 port.
+func configureInterfaceDUT(t *testing.T, dutPort *ondatra.Port, d *oc.Root, desc string) {
+	t.Helper()
+
+	i := d.GetOrCreateInterface(dutPort.Name())
+	i.Description = ygot.String(desc)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+	t.Logf("DUT port %s configured", dutPort)
+}
+
+// generateSubIntfPair configures ATE/DUT SubInterfaces on the target device and returns
+// a slice of the corresponding ATE IPAddresses.
+func generateSubIntfPair(t *testing.T, dut *ondatra.DUTDevice, dutPort *ondatra.Port, ate *ondatra.ATEDevice, atePort *ondatra.Port, top gosnappi.Config, d *oc.Root) []string {
+	t.Helper()
+
+	nextHops := []string{}
+	nextHopCount := 63 // nextHopCount specifies number of nextHop IPs needed.
+	top.Ports().Add().SetName(atePort.ID())
+	for i := 0; i <= nextHopCount; i++ {
+		vlanID := uint16(i) + 1
+		name := fmt.Sprintf(`dst%d`, i)
+		Index := uint32(i) + 1
+		ateIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 1))
+		dutIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 2))
+		configureSubinterfaceDUT(t, d, dutPort, Index, vlanID, dutIPv4)
+		MAC, _ := incrementMAC(atePort1.MAC, i+1)
+		configureATE(t, top, atePort, vlanID, false, name, MAC, dutIPv4, ateIPv4)
+		nextHops = append(nextHops, ateIPv4)
+	}
+	configureInterfaceDUT(t, dutPort, d, "dst")
+	pushConfig(t, dut, dutPort, d)
+	if *deviations.ExplicitInterfaceInDefaultVRF {
+		intf := d.GetOrCreateInterface(dutPort.Name())
+		for i := 1; i <= nextHopCount+1; i++ {
+			fptest.AssignToNetworkInstance(t, dut, intf.GetName(), *deviations.DefaultNetworkInstance, uint32(i))
+		}
+	}
+	return nextHops
+}
+
+// configureSubinterfaceDUT configures a single DUT layer 3 sub-interface.
+func configureSubinterfaceDUT(t *testing.T, d *oc.Root, dutPort *ondatra.Port, index uint32, vlanID uint16, dutIPv4 string) {
+	t.Helper()
+
+	i := d.GetOrCreateInterface(dutPort.Name())
+	s := i.GetOrCreateSubinterface(index)
+	if vlanID != 0 {
+		if *deviations.DeprecatedVlanID {
+			s.GetOrCreateVlan().VlanId = oc.UnionUint16(vlanID)
+		} else {
+			s.GetOrCreateVlan().GetOrCreateMatch().GetOrCreateSingleTagged().VlanId = ygot.Uint16(vlanID)
+		}
+	}
+
+	sipv4 := s.GetOrCreateIpv4()
+
+	if *deviations.InterfaceEnabled && !*deviations.IPv4MissingEnabled {
+		sipv4.Enabled = ygot.Bool(true)
+	}
+	a := sipv4.GetOrCreateAddress(dutIPv4)
+	a.PrefixLength = ygot.Uint8(uint8(ipv4PrefixLen))
+}
+
+// configureATE configures a single ATE layer 3 interface.
+func configureATE(t *testing.T, top gosnappi.Config, atePort *ondatra.Port, vlanID uint16, addPort bool, Name, MAC, dutIPv4, ateIPv4 string) {
+	t.Helper()
+	if addPort {
```

Comment:
Configure ports outside of this function (like in the route delete test) so we don't need the check here.

Reply:
Done